applications/conceptual-model-editor/src/diagram/node/entity-node.tsx

```diff
@@ -89,31 +90,95 @@ function SelectedHighlight() {
+
+  // If canvas toolbar is opened then don't show toolbar
+  if((context?.openedCanvasToolbar ?? null) !== null) {
+    return <></>;
+  }
+
+  if(context?.shouldShowSelectionToolbar()) {
+    return <SelectionToolbar {...props}/>;
+  }
+  else {
+    return <PrimaryNodeToolbar {...props}/>;
+  }
+}
+
+
+function PrimaryNodeToolbar(props: NodeProps<Node<ApiNode>>) {
+  const context = useContext(DiagramContext);
+
   const onShowDetail = () => context?.callbacks().onShowNodeDetail(props.data);
   const onEdit = () => context?.callbacks().onEditNode(props.data);
   const onCreateProfile = () => context?.callbacks().onCreateNodeProfile(props.data);
   const onHide = () => context?.callbacks().onHideNode(props.data);
   const onDelete = () => context?.callbacks().onDeleteNode(props.data);
+  const onAnchor = () => context?.callbacks().onAnchorNode(props.data);
+
   return (
     <>
-      <NodeToolbar position={Position.Top} className="flex gap-2 entity-node-toolbar" >
-        <button onClick={onShowDetail}>ℹ</button>
-        &nbsp;
-        <button onClick={onEdit}>✏️</button>
-        &nbsp;
-        <button onClick={onCreateProfile}>🧲</button>
-        &nbsp;
-      </NodeToolbar>
-      <NodeToolbar position={Position.Right} className="flex gap-2 entity-node-toolbar" >
-        <Handle type="source" position={Position.Right}>🔗</Handle>
-      </NodeToolbar>
-      <NodeToolbar position={Position.Bottom} className="flex gap-2 entity-node-toolbar" >
-        <button onClick={onHide}>🕶</button>
-        &nbsp;
-        <button onClick={onDelete}>🗑</button>
-        &nbsp;
-      </NodeToolbar>
-    </>
+    <NodeToolbar isVisible={props.selected === true} position={Position.Top} className="flex gap-2 entity-node-toolbar" >
+      <button onClick={onShowDetail}>ℹ</button>
+      &nbsp;
+      <button onClick={onEdit}>✏️</button>
+      &nbsp;
+      <button onClick={onCreateProfile}>🧲</button>
+      &nbsp;
+    </NodeToolbar>
+    <NodeToolbar isVisible={props.selected === true} position={Position.Right} className="flex gap-2 entity-node-toolbar" >
+      <Handle type="source" position={Position.Right}>🔗</Handle>
+    </NodeToolbar>
+    <NodeToolbar isVisible={props.selected === true} position={Position.Bottom} className="flex gap-2 entity-node-toolbar" >
+      <button onClick={onHide}>🕶</button>
+      &nbsp;
+      <button onClick={onDelete}>🗑</button>
+      &nbsp;
+      <button onClick={onAnchor}>⚓</button>
+      &nbsp;
+    </NodeToolbar>
+  </>);
+}
+
+
+function SelectionToolbar(props: NodeProps<Node<ApiNode>>) {
+  const context = useContext(DiagramContext);
+  const reactFlow = useReactFlow();
+  const isLastSelected = props.selected === true && context?.getLastSelected() === props.id;
+
+  if (!(isLastSelected === true && context?.getLastSelected() === props.id)) {
+    return null;
+  }
+
+  const onShowSelectionActions = (event: React.MouseEvent) => {
+    // TODO: Don't know where to put this conversion line of code -
```

Comment:
As mentioned somewhere above, should have been marked by something like `TODO PRQuestion:`

And about the amount of TODOs, I know that it may not seem like it, but I really try to minimize the usage of them as possible.